src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/ConsulLongPollCallback.java

```diff
@@ -193,10 +192,17 @@ private long updateIndex(Response response) {
         }
 
         long index = Long.valueOf(indexString);
-        if (currentIndex.get() == index) {
+        long lastIndex = currentIndex.get();
+
+        if (index == lastIndex) {
             stats.indexNotChanged();
             logger.trace("Discarding event on endpoint {} index {} as index did not change", endpoint, index);
             return -1;
+        } else if (index < lastIndex) {
+            stats.indexBackwards();
+            logger.warn("Discarding event on endpoint {} as new index index {} is lower than previous {} - resetting index", endpoint, index, lastIndex);
+            currentIndex.set(0);
```

Comment:
Do we need to set currentIndex to 0?

Reply:
so I set it to 0 actually, but return -1, which signals the caller (`onSuccessfulResponse`)  to ignore the message just received.   TBH the value of `updateIndex` is dubious at most, so perhaps we should remove this function altogether and merge with `onSuccessfulResponse`.  thoughts?